examples/chicane/run_chicane_csr.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022-2023 ImpactX contributors
+# Authors: Marco Garten, Axel Huebl, Chad Mitchell
+# License: BSD-3-Clause-LBNL
+#
+# -*- coding: utf-8 -*-
+
+from impactx import ImpactX, distribution, elements
+
+sim = ImpactX()
+
+# set numerical parameters and IO control
+sim.particle_shape = 2  # B-spline order
+sim.space_charge = False
+# sim.diagnostics = False  # benchmarking
+sim.slice_step_diagnostics = True
+
+# domain decomposition & space charge mesh
+sim.init_grids()
+
+# load a 5 GeV electron beam with an initial
+# normalized transverse rms emittance of 1 um
+kin_energy_MeV = 5.0e3  # reference energy
+bunch_charge_C = 1.0e-9  # used with space charge
+npart = 10000  # number of macro particles
+
+#   reference particle
+ref = sim.particle_container().ref_particle()
+ref.set_charge_qe(-1.0).set_mass_MeV(0.510998950).set_kin_energy_MeV(kin_energy_MeV)
+
+#   particle bunch
+distr = distribution.Gaussian(
+    lambdaX=2.2951017632e-5,
+    lambdaY=1.3084093142e-5,
+    lambdaT=5.5555553e-8,
+    lambdaPx=1.598353425e-6,
+    lambdaPy=2.803697378e-6,
+    lambdaPt=2.000000000e-6,
+    muxpx=0.933345606203060,
+    muypy=0.933345606203060,
+    mutpt=0.999999961419755,
```

Comment:
A lot of the distribution parameters and the accelerator lattice params `rc`, `psi`, `lb` and `ds` are "magic constants".

Since we are in Python here, can we calculate them to make them self-describing?

Reply:
These parameters are set by the Berlin-Zeuthen chicane design.  For a general 4-bend chicane, they are related to the initial beam parameters and to the required final compression factor, and the relations are rather involved.  For testing with CSR, there are some approximate formulas for the predicted emittance growth, but this adds additional complications.  I suggest that this can be revisited later, with low priority.

Reply:
Agreed, thanks for clarifying.